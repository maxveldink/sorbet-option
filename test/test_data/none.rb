# typed: strict
# frozen_string_literal: true

class TestGenerics
  extend T::Sig

  sig { params(should_be_some: T::Boolean).returns(Typed::Option[Integer]) }
  def test_generic_initializer(should_be_some)
    if should_be_some
      Typed::Some.new(1)
    else
      Typed::None.new
    end
  end

  sig { void }
  def test_payload_or
    none = Typed::None.new

    none.payload_or(1).upcase
  end
end
