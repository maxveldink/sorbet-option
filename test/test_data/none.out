test/test_data/none.rb:20: Method `upcase` does not exist on `Integer` https://srb.help/7003
    20 |    none.payload_or(1).upcase
                               ^^^^^^
  Got `Integer` originating from:
    test/test_data/none.rb:20:
    20 |    none.payload_or(1).upcase
            ^^^^^^^^^^^^^^^^^^
  Did you mean `phase`? Use `-a` to autocorrect
    test/test_data/none.rb:20: Replace with `phase`
    20 |    none.payload_or(1).upcase
                               ^^^^^^
    https://github.com/sorbet/sorbet/tree/19c765c0bbcf09e0e545a50535c6ec0c94690b76/rbi/core/integer.rbi#L921: Defined here
     921 |  def phase(); end
            ^^^^^^^^^^^
